<?xml version="1.0" encoding="utf-8"?>
<launch>
	<master auto="start"/>

	<arg name="rviz" default="true"/>

	<!-- Set static tf transformations -->
	<node pkg="tf" type="static_transform_publisher" name="map_broadcaster" args="0 0 0 0 0 0 1 map odom 10"/>
        <param name="robot_description" textfile="$(find pses_simulation)/data/urdf_model/pses_robot_simple.urdf"/>
	<node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher"/>
	<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher"/>
	<!-- Run the necessary nodes to bring up the car -->
        <node pkg="pses_simulation" type="odom_sim" name="odom_sim" output="screen"/>
        <node pkg="pses_simulation" type="us_sim" name="us_sim" output="screen"/>
        <node pkg="pses_simulation" type="laser_sim" name="laser_sim" output="screen"/>
        <node pkg="pses_simulation" type="virtual_info" name="virtual_info" output="screen">
        	<param name="car_id" value="0"/>
        </node>
	<!-- Run the map server -->
        <node name="map_server" pkg="map_server" type="map_server" args="$(find pses_simulation)/data/map/map.yaml"/>

	<group if="$(arg rviz)">
                <node pkg="rviz" type="rviz" name="rviz" args="-d $(find pses_simulation)/data/viewer/viewer.rviz"/>
	</group>
</launch>
